firebase-firestore/src/main/java/com/google/firebase/firestore/FirestoreRegistrar.java

```diff
@@ -41,13 +44,17 @@
         Component.builder(FirestoreMultiDbComponent.class)
             .add(Dependency.required(FirebaseApp.class))
             .add(Dependency.required(Context.class))
+            .add(Dependency.requiredProvider(HeartBeatInfo.class))
+            .add(Dependency.requiredProvider(UserAgentPublisher.class))
             .add(Dependency.optional(InternalAuthProvider.class))
             .factory(
                 c ->
                     new FirestoreMultiDbComponent(
                         c.get(Context.class),
                         c.get(FirebaseApp.class),
-                        c.get(InternalAuthProvider.class)))
+                        c.get(InternalAuthProvider.class),
+                        new FirebaseClientGrpcMetadataProvider(
+                            c.get(UserAgentPublisher.class), c.get(HeartBeatInfo.class))))
```

Comment:
This needs to change to `c.getProvider()`, and also `FirebaseClientGrpcMetadataProvider` needs to change to use `Provider<T>` instead of values directly